src/core/public/http/external_url_service.ts

```diff
@@ -50,20 +50,33 @@ function normalizeProtocol(protocol: string) {
   return protocol.endsWith(':') ? protocol.slice(0, -1).toLowerCase() : protocol.toLowerCase();
 }
 
+const createIsInternalUrlValidation = (
+  location: Pick<Location, 'href'>,
+  serverBasePath: string
+) => {
+  return function isInternallUrl(next: string) {
+    const base = new URL(location.href);
```

Comment:
nit: this can be created one level up to avoid repeating the operation on every `isInternalUrl` call 😇

Reply:
I applied your suggestion also to `createExternalUrlValidation`